### DISCUSSION TITLE
Return type for data useContractRead

### DISCUSSION THREAD

**Discussion by validateITNikita** [unicape/use-wagmi#60] [Q&A] [answered]
Is it possible to set return type for contract read?

`const { data, refetch } = useContractRead(config);`

for now data is `ref<undefined>`

how to make it this, for example:

`const { data, refetch } = useContractRead<BigNumberish>(config);`

type of data is `ref<BigNumberish> `

**Comment by unicape** [comment]
@validateITNikita The return type is derived from abi

**Reply by validateITNikita** [reply]
Thank you for reply!
Could you help me to understand it.
I generate abi via wagmi cli.
copied this abi and used it in config this way

```
export const readTotalSupplyConfig: UseContractReadConfig = {
    address: useGetContractAddress(),
    abi: ABI,
    functionName: "totalSupply",
}
```

after that i use it in composable

`const { data: totalSupply, refetch: refetchTotalSupply } = useContractRead(readTotalSupplyConfig);`

type of data is undefined

this is part of ABI

```
{
    stateMutability: 'view',
    type: 'function',
    inputs: [],
    name: 'totalSupply',
    outputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
  },
  ```
  
Thank you for any help and for this great wagmi library for vue / nuxt 🔆

**Reply by unicape** [reply, marked as answer]
@validateITNikita `readTotalSupplyConfig` Try writing it directly into `useContractRead`

**Reply by NikitaVelixJob** [reply]
Thank you! This worked! 😍
Why though? Interesting.
